src/tests/src/solver/simulation/tests-ts-numbers.cpp

```diff
@@ -751,3 +751,39 @@ BOOST_AUTO_TEST_CASE(check_all_drawn_ts_numbers_are_bounded_between_0_and_nb_of_
 	BOOST_CHECK(thermalTsNumber < thermalNumberOfTs);
     BOOST_CHECK_LT(binding_constraints_TS_number, binding_constraints_number_of_TS);
 }
+
+BOOST_AUTO_TEST_CASE(split_string_ts_cluster_gen)
```

Comment:
You've tested the string -> token mapping. You need to test also the token -> ThermalCluster lookup. Create a study with a few clusters and test that they are found if they exist, not found if they don't, etc.